lesson_16/api/api_app/src/main/java/com/codedifferently/lesson16/web/PatronsController.java

```diff
@@ -0,0 +1,60 @@
+package com.codedifferently.lesson16.web;
+
+import com.codedifferently.lesson16.library.Library;
+import com.codedifferently.lesson16.library.LibraryGuest;
+import com.codedifferently.lesson16.library.exceptions.MediaItemCheckedOutException;
+import com.codedifferently.lesson16.library.search.SearchCriteria;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/patrons")
+public class PatronsController {
+
+  private final Library library;
+
+  public PatronsController(Library library) {
+    this.library = library;
+  }
+
+  @GetMapping
+  public ResponseEntity<Collection<LibraryGuest>> getPatrons() {
+    Collection<LibraryGuest> patrons = library.getPatrons();
+    return ResponseEntity.ok(patrons);
+  }
+
+  @PostMapping
+  public ResponseEntity<?> createPatron(@RequestBody LibraryGuest patron) {
```

Comment:
You should not expose the `LibraryGuest` interface to the outside world. Please create proper request/response entities following the pattern I've established for you (here and below).